les/client_handler.go

```diff
@@ -45,6 +46,47 @@ type clientHandler struct {
 	closeCh  chan struct{}
 	wg       sync.WaitGroup // WaitGroup used to track all connected peers.
 	syncDone func()         // Test hooks when syncing is done.
+
+	//@rayyuan
+	gatewayFeeCache *gatewayFeeCache
+}
+
+type gatewayFeeEtherbase struct {
+	GatewayFee *big.Int
+	Etherbase  common.Address
+}
+
+func newGatewayFeeEtherbase(gatewayFee *big.Int, etherbase common.Address) *gatewayFeeEtherbase {
+	ret := &gatewayFeeEtherbase{
+		GatewayFee: gatewayFee,
+		Etherbase:  etherbase,
+	}
+	return ret
+}
+
+type gatewayFeeCache struct {
+	mux           *sync.RWMutex
```

Comment:
The word "mux" means something specific and different. Calling this "mutex" or "lock" is better.